### DISCUSSION TITLE
What's the cool thing I will be able to do in maizzle 4.0 :)

### DISCUSSION THREAD

**Discussion by weenzeel** [maizzle/framework#639] [Q&A] [answered]
I saw this tweet:

<img width="524" alt="image" src="https://user-images.githubusercontent.com/9976890/156823027-78de7c1d-4458-4c6b-8575-bc6fb97df481.png">

What is the new cool thing? I don't know Maizzle all that well.

**Comment by cossssmin** [comment, marked as answer]
You can pass data through attributes to a component.

In that screenshot, `article` inside `<component ... role="article">` is passed into the component, compiling `role="{{ role }}"` to `role="article"`.

Might not be the best example since I used `article` for the component file name as well (`article.html`).

Here's a simpler component example:

```html
<!-- button.html -->
<a href="{{ url }}">
  {{ label }}
</a>
```

If you'd use that like this:

```html
<component
  src="button.html"
  url="https://example.com"
  label="Click me"
></component>
```

... it would compile to:

```html
<a href="https://example.com">
  Click me
</a>
```

**Reply by HelloAlexPan** [reply]
Is there a typo?

Currently it reads

`<div class="{{ class }}`

Should it be

`<div style="{{ class }}"`

**Reply by cossssmin** [reply]
Not a typo, it's being passed Tailwind classes and the result has CSS inlined.
